tests/settings/ConfigurationGatheringTest.py

```diff
@@ -222,6 +224,53 @@ def test_targets(self):
 
         self.assertEqual(targets, ['cli', 'test1', 'test2'])
 
+    def test_targets_substring(self):
+        current_dir = os.path.abspath(os.path.dirname(__file__))
+        test_dir = os.path.join(current_dir, 'section_manager_test_files')
+        with change_directory(test_dir):
+            log_printer = ListLogPrinter()
+            _, _, _, targets = gather_configuration(
+                lambda *args: True,
+                log_printer,
+                arg_list=['-c', 'inherit_coafile', 'python'])
+            self.assertEqual(
+                targets, ['python', 'all.python', 'all.python.codestyle'])
```

Comment:
I am curious why we want `python` to be a target, when it isnt a value section of the `.coafile`.

It feels like that might cause other problems. cc @Udayan12167 who will know the ramifications of this better.